demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -155,3 +169,23 @@ def add_invalid_content_items(self):
                 for invalid_path in self.invalid_items
             ]
         )
+
+    def filter_validation_results(
+        self, validation_results: List[ValidationResult]
+    ) -> List[ValidationResult]:
+        """
+        Filters out validation results for error codes that are both in ALWAYS_RUN_ON_ERROR_CODE
+        and in the content object's ignored_errors list.
+        This addresses unique cases where a validation must run first, then filter the relevant results afterward based on the results.
+
+        Returns:
+        List[ValidationResult]: Filtered validation results excluding ignored error codes
+        """
+        return [
+            result
+            for result in validation_results
+            if not (
+                result.validator.error_code in ALWAYS_RUN_ON_ERROR_CODE
+                and result.validator.error_code in result.content_object.ignored_errors
```

Comment:
why are we checking the first condition? if you got here it means we're already in an error code that is in ALWAYS_RUN_ON_ERROR_CODE isn't it?

Reply:
I think these are similar looking but different checks.
The initial check in the if statement:
```python
if any(
  error_code in ALWAYS_RUN_ON_ERROR_CODE
  for error_code in {
    validator.error_code for validator in self.validators
  }
):
```
Is just an initial indication that **at least** one validator has an error code in `ALWAYS_RUN_ON_ERROR_CODE`.
Since this code is in the main `run_validation` functionality, other validators can run along side GR107.

The code you mentioned in the `filter_validation_results` function actually goes over every `ValidationResult` object and filters according according to what is needed, and `result.validator.error_code in ALWAYS_RUN_ON_ERROR_CODE` is one of the necessary filtering conditions.

Reply:
Fixed.